sonar-python-plugin/src/test/java/org/sonar/plugins/python/SymbolVisitorTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.sonar.api.batch.fs.TextPointer;
+import org.sonar.api.batch.fs.TextRange;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.fs.internal.DefaultTextPointer;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.python.PythonVisitorContext;
+import org.sonar.python.TestPythonVisitorRunner;
+import org.sonar.python.api.tree.FileInput;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SymbolVisitorTest {
+
+  private static SensorContextTester context;
+  private static String componentKey;
+
+  @BeforeClass
+  public static void scanFile() {
+    String dir = "src/test/resources/org/sonar/plugins/python/squid-sensor";
+
+    File file = new File(dir, "/symbolVisitor.py");
+    DefaultInputFile inputFile = TestInputFileBuilder.create("moduleKey", file.getName())
+      .initMetadata(TestUtils.fileContent(file, StandardCharsets.UTF_8))
+      .build();
+
+    context = SensorContextTester.create(file);
+    context.fileSystem().add(inputFile);
+    componentKey = inputFile.key();
+
+    SymbolVisitor symbolVisitor = new SymbolVisitor(context.newSymbolTable().onFile(inputFile));
+    PythonVisitorContext context = TestPythonVisitorRunner.createContext(file);
+    FileInput fileInput = context.rootTree();
+    fileInput.accept(symbolVisitor);
+  }
+
+  @Test
+  public void symbol_visitor() {
+    assertThat(context.referencesForSymbolAt(componentKey, 1, 10)).isNull();
+    verifyUsages(componentKey, 1, 0);
```

Comment:
I think you could remove componentKey as parameter of this method and use the field directly.